students/Gregg/lesson3/mailroom.py

```diff
@@ -151,9 +152,15 @@ def create_donation_report():
     row_list = []
     row_list.append(format_column_header())
     row_list.append('_' * 68)
-    for donor in donations:
+    donor_rows_list = []
+    hisorical_donation_amount_list = []
+    for donor in donors:
         donor_stats = get_row(donor)
-        row_list.append(format_row(donor_stats))
+        hisorical_donation_amount_list.append(donor_stats[1])
+        donor_rows_list.append(format_row(donor_stats))
+    hisorical_donation_amount_list, donor_rows_list = zip(*sorted(zip(hisorical_donation_amount_list, donor_rows_list), reverse=True))
```

Comment:
this would be much simpler as a list of lists to sort and then use for report generation